snuba/consumer.py

```diff
@@ -86,27 +78,31 @@ def delivery_callback(self, error, message):
     def flush_batch(self, batch: Sequence[ProcessedMessage]):
         """First write out all new INSERTs as a single batch, then reproduce any
         event replacements such as deletions, merges and unmerges."""
-        inserts = []
-        replacements = []
+        inserts: MutableSequence[WriterTableRow] = []
+        replacements: MutableSequence[ReplacementBatch] = []
 
-        for message in batch:
-            if message.action == ProcessorAction.INSERT:
-                inserts.extend(message.data)
-            elif message.action == ProcessorAction.REPLACE:
-                replacements.extend(message.data)
+        for item in batch:
+            if isinstance(item, InsertBatch):
+                inserts.extend(item.rows)
+            elif isinstance(item, ReplacementBatch):
+                replacements.append(item)
+            else:
+                raise TypeError(f"unexpected type: {item}")
```

Comment:
Should this be `f"Unexpected type: {type(item)}"` so we see the type even if item provides repr and str methods that hide the type.

Reply:
I'd hope that we include the type name as part of `__repr__` if it's not a built-in but I changed it just to be extra safe: 
82383c3